functions/create_draft/interactivity_handler.ts

```diff
@@ -65,14 +67,62 @@ export const openDraftEditView: BlockActionHandler<
     if (!viewsOpenResp.ok) {
       const draftEditModalErrorMsg =
         `Error opening up the draft edit modal view. Contact the app maintainers with the following - (Error detail: ${viewsOpenResp.error}`;
-      console.log(draftEditModalErrorMsg);
+      console.error(draftEditModalErrorMsg);
 
       await client.functions.completeError({
         function_execution_id: body.function_data.execution_id,
         error: draftEditModalErrorMsg,
       });
     }
   }
+
+  // If the user selects to discard the draft message
+  if (action.selected_option.value === "discard_message_overflow") {
+    const id = action.block_id;
+    const message_ts = body.message?.ts || "";
+
+    try {
+      // Delete the draft from the 'drafts' Datstore
+      const deleteResp = await client.apps.datastore.delete({
+        datastore: DraftDatastore.name,
+        id: id,
+      });
+
+      // If the DELETE datastore record operation fails, log the error and complete the function with an error
+      if (!deleteResp.ok) {
+        const deleteDraftErrorMsg =
+          `Error deleting draft with id ${id}. Contact the app maintainers with the following - (Error detail: ${deleteResp.error})`;
+        console.error(deleteDraftErrorMsg);
+
+        await client.functions.completeError({
+          function_execution_id: body.function_data.execution_id,
+          error: deleteDraftErrorMsg,
+        });
+      }
```

Comment:
I think here we can `throw` this error message and let the `catch` handle the `console.error` and `client.functions.completeError` logic! This should let us return from the error state without deleting the chat message ✉️ 💨

Reply:
Thanks so much for helping me along here!  TypeScript/Deno is definitely new territory for me :-D 
Let me know if this latest commit is sufficient.